app/main.py

```diff
@@ -1,2 +1,52 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = 8) -> None:
+        self.capacity = capacity
+        self.size = 0
+        self.hash_table = [[] for _ in range(self.capacity)]
+
+    def calculate_hash(self, key: Any) -> int:
+        return hash(key) % self.capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        index = self.calculate_hash(key)
+        container = self.hash_table[index]
+
+        for index, (key_in_container, _) in enumerate(container):
+            if key_in_container == key:
+                container[index] = (key, value)
+                return
+
+        container.append((key, value))
```

Comment:
You should change the entire concept of this. Storing values in a list of lists makes your code very slow and we are trying to achieve maximum performance. Try to follow the implementation explained in course materials.